### PR TITLE
Allow use of `%{name}` named references in format strings.

### PR DESCRIPTION
**Pull request by FeepingCreature**
They are just treated identically to %s.

Precursor PR to JSON encoding support, so we can start using `%{name}` in libraries.

### CODE REVIEW

**Review by linkrope — CHANGES_REQUESTED** [review]

**Comment by linkrope** [comment]
I moved the repository to funkwerk-mobility.

File: src/util/log.d [hunk 1]
```diff
@@ -183,6 +185,27 @@ struct Log
...
 
+// Helper to allow using named references in format strings, ie. treat %{name} equivalent to %s.
+private string bracketsToFormatCode(string fmt)
+{
+    string result = fmt;
+
+    while (result.canFind("%{"))
+    {
+        enforce(result.find("%{").canFind("}"), fmt ~ ": terminating '}' not found");
+        result = chain(result.until("%{"), "%s", result.find("%{").find("}").dropOne).toUTF8;
```
[the remark this answers is not in the document]
Comment (FeepingCreature) [reply src/util/log.d:196]: `chain` forces dchar.

File: src/util/log.d [hunk 2]
```diff
@@ -183,6 +185,27 @@ struct Log
     }
 }
 
+// Helper to allow using named references in format strings, ie. treat %{name} equivalent to %s.
```
[the remark this answers is not in the document]
Comment (FeepingCreature) [reply src/util/log.d:188]: Allgemein vermeid ich std.regex lieber.

[the remark this answers is not in the document]
Comment (FeepingCreature) [reply src/util/log.d:188]: Kommentar added.

[the remark this answers is not in the document]
Comment (FeepingCreature) [reply src/util/log.d:188]: renamed `simplifyNamedReferences`

File: src/util/log.d (hunk 1 above)
[the remark this answers is not in the document]
Comment (FeepingCreature) [reply src/util/log.d:197]: Replaced with `~`.

File: src/util/log.d [hunk 3]
```diff
@@ -183,6 +185,28 @@ struct Log
...
+// Avoid std.regex because it's used at CTFE.
+private string simplifyNamedReferences(string fmt)
+{
+    string result = fmt;
+
+    while (result.canFind("%{"))
+    {
+        enforce(result.find("%{").canFind("}"), fmt ~ ": terminating '}' not found");
+        result = result[0 .. result.countUntil("%{")] ~ "%s" ~ result.find("%{").find("}").dropOne;
```
Comment (linkrope) [inline src/util/log.d:197]: result = result[0 .. $ - result.find("%{").length] ~ ...

just to get along with `find` only.

File: src/util/log.d [hunk 4]
```diff
@@ -9,12 +9,14 @@ import std.algorithm;
...
 import std.conv;
 import std.datetime;
+import std.exception : enforce;
 import std.format;
 import std.range;
 import std.stdio;
 import std.string;
 import std.traits;
 import std.typecons;
+import std.utf : toUTF8;
```
Comment (linkrope) [inline src/util/log.d:19]: obsolete?

File: src/util/log.d [hunk 5]
```diff
@@ -399,7 +423,6 @@ version (Posix)
         {
             import core.stdc.signal : signal, SIG_ERR;
             import core.sys.posix.signal : SIGHUP;
-            import std.exception : enforce;
```
Comment (linkrope) [inline src/util/log.d:402]: If you keep this local import and repeat the import locally in the helper function?
Then it's easier to get rid of obsolete imports.

Comment (FeepingCreature) [reply src/util/log.d]: done

File: src/util/log.d (hunk 4 above)
Comment (FeepingCreature) [reply src/util/log.d]: removed

File: src/util/log.d [hunk 6]
```diff
@@ -183,6 +185,28 @@ struct Log
     }
 }
 
+// Helper to allow using named references in format strings, ie. treat %{name} equivalent to %s.
+// Avoid std.regex because it's used at CTFE.
+private string simplifyNamedReferences(string fmt)
+{
+    string result = fmt;
```
[the remark this answers is not in the document]
Comment (FeepingCreature) [reply src/util/log.d]: done

File: src/util/log.d (hunk 3 above)
Comment (FeepingCreature) [reply src/util/log.d]: done

**Comment by FeepingCreature** [comment]
I think you need to merge...

**Pull request merged**